compaction/compaction_manager.cc

```diff
@@ -1270,7 +1289,7 @@ class compaction_manager::rewrite_sstables_compaction_task : public compaction_m
             auto run_identifier = sst->run_identifier();
             // FIXME: this compaction should run with maintenance priority.
             auto descriptor = sstables::compaction_descriptor({ sst }, service::get_local_compaction_priority(),
-                sstable_level, sstables::compaction_descriptor::default_max_sstable_bytes, run_identifier, _options);
+                sstable_level, sstables::compaction_descriptor::default_max_sstable_bytes, run_identifier, _options, _owned_ranges_ptr);
 
```

Comment:
Then the owned ranges ptr specified in the descriptor by a specific compaction type won't be overriden unnecessarily.

Reply:
done in v4